editoast/src/views/timetable.rs

```
@@ -367,6 +375,77 @@ pub struct ElectricalProfileSetIdQueryParam {
+    pub work_schedule_ids: Vec<i64>,
+    /// Datetime of the start of the conflict
+    pub start_time: DateTime<Utc>,
+    /// Datetime of the end of the conflict
+    pub end_time: DateTime<Utc>,
+    /// Type of the conflict
+    #[schema(inline)]
+    pub conflict_type: ConflictType,
+    /// List of requirements causing the conflict
+    pub requirements: Vec<ConflictRequirement>,
+}
+
+impl Conflict {
+    /// Converts the conflict data into a `ViewConflict` format.
+    ///
+    /// This function processes train schedule ids and paced trains generated ids ("paced_train_id#occurrence_id}")
+    ///  and maps them to either a `train_schedule_id` or a `paced_train_occurrence_id` based on the provided key mapping.
+    /// The train_id_map follows this structure:
+    /// - `train_id: (train_schedule_id, None)` if `train_id` corresponds to a train schedule ID.
+    /// - `train_id: (paced_train_id, occurrence_id)` if `train_id` corresponds to a generated paced train ID.
+    fn into_conflict_response(
+        conflict: CoreConflict,
+        train_id_map: HashMap<String, TrainId>,
+    ) -> Self {
+        let mut train_schedule_ids = Vec::new();
+        let mut paced_train_occurrence_ids = Vec::new();
+
+        for train_id in &conflict.train_ids {
+            if let Some(train_id) = train_id_map.get(train_id) {
+                match train_id {
+                    TrainId::TrainSchedule(train_id) => train_schedule_ids.push(*train_id),
+                    TrainId::PacedTrainOccurrence(paced_train_occurrence) => {
+                        paced_train_occurrence_ids.push(paced_train_occurrence.clone())
+                    }
+                }
+            } else {
+                let train_id = train_id
+                    .parse::<i64>()
+                    .unwrap_or_else(|_| panic!("Failed to parse train_id '{}'", train_id));
+                train_schedule_ids.push(train_id);
+            }
```

This case is weird. If core sends a train ID that does not match any train ID provided in input, then something is wrong.  In this case, I would throw an error, or at least log a warning and skip it.

See this comment: https://github.com/OpenRailAssociation/osrd/pull/11069#discussion_r1995170230